### ISSUE TITLE
IE recursive loop crash.

### ISSUE THREAD

**Issue by blizznet** [JillElaine/jquery-idleTimeout#40]
Chrome, FireFox are fine with the multiple tabs (and DevEx MVC).. Works as expected

IE on the other hand (with DevEx MVC Extensions) blows up with more than 3 open tabs.

JavaScript critical error in (unknown source location)\n\nSCRIPT7: Out of memory
It appears to have a recursive looping problem after 2 or 3 tabs are redirected to logout.
Anyone know why, or any way to fix this ?

**Comment by rhetherington** [comment]
I believe this version built upon JillElaines is more stable and also accommodates both JQuery UI and Bootstrap UI's.

https://www.npmjs.com/package/jquery-idleTimeout-plus
https://github.com/LinearSoft/jquery-idleTimeout-plus

Hope this helps.

Regards,


Richard Hetherington
Senior Web Developer

Aztec Retail Solutions
c/o John Smith & Son Group Ltd
Dragon House
Headlands Business Park
Ringwood,
Hampshire
BH24 3PB
Tel: +44 (0)1425 485957 ext. 315
Fax: +44 (0)1425 489857

**Comment by blizznet** [comment]
Thanks, could not get the multiple tabs to function in those versions.

**Comment by rhetherington** [comment]
Yeah, you are quite correct. It appears that the users choice to stay logged in doesn't work across tabs. Sorry, I only just incorporated this plus version in a new project and haven't as yet given it the full suite of tests. It looked good off the shelf, however I forgot to test the multi tab instances. My bad. Perhaps it's a quick fix for the developers.

Regards,


Richard Hetherington
Senior Web Developer

Aztec Retail Solutions
c/o John Smith & Son Group Ltd
Dragon House
Headlands Business Park
Ringwood,
Hampshire
BH24 3PB
Tel: +44 (0)1425 485957 ext. 315
Fax: +44 (0)1425 489857

Richard

**Comment by JillElaine** [comment]
Several users have reported problems with Internet Explorer & asp apps, and users have also reported problems with 'redirect' firing multiple times, which may be related. Please look at Issues #21, #26 & #38. I apologize that I do not have time to work on my script right now. Please report back if you find a fix.

**Comment by rhetherington** [comment]
Hello again, after reviewing all of the jquery-idleTimeout-plus configuration, I discovered that multiWindowSupport config param needs set in your initialisation for the plugin. It is not enabled by default for some odd reason. In order for this to work you also need to include the storage API script on your pages also.

Here's my setup. Paths to scripts will vary. I have removed all version numbers to simplify example.


<script src="/js/jquery.min.js" type="text/javascript"></script>

<script src="/js/theme-dashboard/toolkit.min.js" type="text/javascript"></script>

<script src="/js/jquery-idleTimeout-plus-iframe.min.js" type="text/javascript"></script>

<script src="/js/jquery.storageapi.min.js" type="text/javascript"></script>

<script src="/js/jquery.blockUI.min.js" type="text/javascript"></script>

$(document).ready(function() {
                // https://www.npmjs.com/package/jquery-idleTimeout-plus
                // https://github.com/LinearSoft/jquery-idleTimeout-plus
                IdleTimeoutPlus.start({
                                idleTimeLimit: 60 * 30,                                                                                   // 'No activity' time limit - NOW in SECONDS. 1800 = 30 Minutes
                                warnTimeLimit: 60,                                                                                         // Length of time to display warning
                                multiWindowSupport: true,
                                bootstrap: true,                                                                                                // works with JQuery UI OR Bootstrap
                                redirectUrl: my-account/logout?action=AutoLogOut",                   // redirect to this url if timeout exceeded in warning dialog
                                logoutUrl: "my-account/logout",                                                              // If user clicks logout in warning dialog
                                warnTitle: "WARNING",
                                warnMessage: "You are about to be signed out due to inactivity."
                });
});

This now works perfectly for me across multiple tabs/windows and inside iframes of the same origin.

Hope this helps.

Regards,



Richard Hetherington
Senior Web Developer

Aztec Retail Solutions
c/o John Smith & Son Group Ltd
Dragon House
Headlands Business Park
Ringwood,
Hampshire
BH24 3PB
Tel: +44 (0)1425 485957 ext. 315
Fax: +44 (0)1425 489857

**Comment by blizznet** [comment]
Thanks. I just came up with a workaround solution for IE with the JillElaine version, and store.js

 

We have application that opens up multi tabs via java script, I was able to avoid the looping issue by simply

closing the tabs that the main page creates, on the time out, then logging out, the last open Tab from that session(domain) .

 This way I only have one window to “Log out” of and the script works ok , in fact the way it was, it would handle up to 3 

total tabs open in IE 11 – Win 7 64, many more in other browsers. Not tested in Edge or Win 10 at all.

One of the problems is this app is DevExpress heavy.. and the way DevEx MVC extensions handle JavaScript causes issues.

Especially in IE.

 

 

Thanks for the follow up, much appreciated.

**Comment by zacharymehl** [comment]
I could not get the multiple tabs to work in Edge or IE 10. I tried jquery-idleTimeout-plus and was still seeing the same issue. My resolution was to install [localForage,](https://github.com/localForage/localForage) which uses an indexedDB instead of local storage and replace the calls using store.js with localForage.js
